Source/Flow/Private/FlowAsset.cpp

```diff
@@ -421,34 +470,76 @@ void UFlowAsset::FinishFlow(const EFlowFinishPolicy InFinishPolicy, const bool b
 	}
 }
 
+bool UFlowAsset::HasStartedFlow() const
+{
+	return RecordedNodes.Num() > 0;
+}
+
+AActor* UFlowAsset::TryFindActorOwner() const
+{
+	UActorComponent* OwnerAsComponent = Cast<UActorComponent>(GetOwner());
+	if (IsValid(OwnerAsComponent))
+	{
+		return Cast<AActor>(OwnerAsComponent->GetOwner());
+	}
+
+	return nullptr;
+}
+
 TWeakObjectPtr<UFlowAsset> UFlowAsset::GetFlowInstance(UFlowNode_SubGraph* SubGraphNode) const
 {
 	return ActiveSubGraphs.FindRef(SubGraphNode);
 }
 
-void UFlowAsset::TriggerCustomEvent(UFlowNode_SubGraph* Node, const FName& EventName) const
```

Comment:
Because I changed around the custom input/output event calls, I made them more clear which versions were subgraph-specific and which are called for root graphs as well.

Previously, CustomOutputs wouldn't get called on a root graph, now they are, so some disambiguation on the function naming was necessary to keep the code clear.

Reply:
Sure. I admit I'm puzzled about what would be the best name for the function calling Custom Event from SubGraph... For now, I'm settled on `TriggerCustomInput_FromSubGraph`. Let me know if my version sounds worse to your ears ;)

Reply:
Sounds fine.  I just needed to disambiguate between the subgraph version and the general (root or subgraph) version.